README

```diff
@@ -53,11 +53,33 @@ You can also use the image standalone (without TFTP under Foreman's
 control). In this case, edit your pxelinux.cfg/default file directly and
 make sure the foreman.url points correctly.
 
+Extending the image at runtime with a ZIPfile
+---------------------------------------------
+
+It's possible to provide a zip containing extra code for the image to use.
+First, create a directory structure like:
+
+```
+./bin
+./facts
+./lib
+./lib/ruby
+```
+
+`bin` is added to PATH, you can place binaries here. If this directory
+contains an executable `autostart` script, it will be run *before* the
+image is registered to Foreman. `facts` is added to FACTERLIB so that
+custom facts may be configured and sent to Foreman. `lib` is added to
+LD_LIBRARY_PATH and `lib/ruby` is added to RUBYLIB, so that binaries in
+`bin` can be executed properly.
+
+Currently, the location of this is hardcoded - the image will look for
+`#{foreman.url}/extension.zip` so place the zip in your Foreman public/
+directory (TODO, add proper routes to the proxy for this).
```

Comment:
My initial idea was to accept the following kernel command line options:

    ext=http://any.server.you.like/ext1.zip ext=http://any.server.you.like/ext2.zip ext=http://any.server.you.like/ext3.zip

Since we get the command line verbosely, we can simply allow multiple instances of the same option and read that properly. Or another possibility would be:

    extbase=http://any.server.you.like/extensions ext=ext1.zip ext=ext2.zip